src/sparse/impl/KokkosSparse_spgemm_impl_speed.hpp

```diff
@@ -314,6 +313,21 @@ struct KokkosSPGEMM
           shmem_key_size = shmem_key_size + ((shmem_key_size - shmem_hash_size) * sizeof(nnz_lno_t)) / (sizeof (nnz_lno_t) * 2 + sizeof(scalar_t));
           shmem_key_size = (shmem_key_size >> 1) << 1;
 
+          // Add check that memory is partitioned into aligned chunks
+          nnz_lno_t remainder_memory = thread_memory - sizeof(nnz_lno_t)*2 - shmem_hash_size*sizeof(nnz_lno_t);
+          // The remainder of memory is for vals, must be aligned into sizeof(scalar_t) chunks, and there must be at least as many entries as keys
+          nnz_lno_t val_memory = remainder_memory - 2*shmem_key_size*sizeof(nnz_lno_t);
+
+          while (val_memory % sizeof(scalar_t) > 0)
+          {
+            if ((val_memory/sizeof(scalar_t) - shmem_key_size) / shmem_key_size > 1)
+              shmem_key_size += 1;
+            else
```

Comment:
Same question as before. Is there any way that `remainder_memory` is an odd number? Again I think we get an infinite loop if `val_memory` is odd and `sizeof(scalar_t)` is even.

Reply:
@jjwilke I rushed this in, need to add safety though I don't think this could happen based on the types that are used to select these values.